x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/rules_table/rules_table_reducer.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type React from 'react';
+import { EuiBasicTable } from '@elastic/eui';
+import { FilterOptions, PaginationOptions, Rule } from '../types';
+
+export type LoadingRuleAction =
+  | 'load'
+  | 'duplicate'
+  | 'enable'
+  | 'disable'
+  | 'export'
+  | 'delete'
+  | null;
+
+export interface RulesTableState {
+  rules: Rule[];
+  pagination: PaginationOptions;
+  filterOptions: FilterOptions;
+  loadingRulesAction: LoadingRuleAction;
+  loadingRuleIds: string[];
+  selectedRuleIds: string[];
+  exportRuleIds: string[];
+  lastUpdated: number;
+  isRefreshOn: boolean;
+  showIdleModal: boolean;
+}
+
+export type RulesTableAction =
+  | { type: 'setRules'; rules: Rule[]; pagination: Partial<PaginationOptions> }
+  | { type: 'updateRules'; rules: Rule[] }
+  | {
+      type: 'updateFilterOptions';
+      filterOptions: Partial<FilterOptions>;
+      pagination: Partial<PaginationOptions>;
+    }
+  | { type: 'loadingRuleIds'; ids: string[]; actionType: LoadingRuleAction }
+  | { type: 'selectedRuleIds'; ids: string[] }
+  | { type: 'exportRuleIds'; ids: string[] }
+  | { type: 'setLastRefreshDate' }
+  | { type: 'setAutoRefreshOn'; on: boolean }
+  | { type: 'setShowIdleModal'; show: boolean }
+  | { type: 'failure' };
+
+export const createRulesTableReducer = (
+  tableRef: React.MutableRefObject<EuiBasicTable<unknown> | undefined>
+) => {
+  const rulesTableReducer = (state: RulesTableState, action: RulesTableAction): RulesTableState => {
+    switch (action.type) {
+      case 'setRules': {
+        if (
+          tableRef != null &&
+          tableRef.current != null &&
+          tableRef.current.changeSelection != null
+        ) {
```

Comment:
can we consolidate this into just

```typescript
if(tableRef?.current?.changeSelection != null){...
```

Reply:
I'll do it in https://github.com/elastic/kibana/pull/91925, thank you.